tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/OCLCodeCache.java

```diff
@@ -558,6 +557,57 @@ private boolean isInputSourceSPIRVBinary(byte[] source) {
         return value == SPIRV_MAGIC_NUMBER;
     }
 
+    private void dumpKernelSource(String id, String entryPoint, String log, byte[] source) {
+        final Path outDir = resolveLogDirectory();
+        final String identifier = STR."\{id}-\{entryPoint}";
+        error("Unable to compile task %s: check logs at %s/%s.log", identifier, outDir.toAbsolutePath(), identifier);
+
+        File file = new File(STR."\{outDir}/\{identifier}.log");
+        try (FileOutputStream fos = new FileOutputStream(file)) {
+            fos.write(log.getBytes());
+        } catch (IOException e) {
+            error("unable to write error log: ", e.getMessage());
+        }
+        file = new File(STR."\{outDir}/\{identifier}\{OPENCL_SOURCE_SUFFIX}");
+        try (FileOutputStream fos = new FileOutputStream(file)) {
+            fos.write(source);
+        } catch (IOException e) {
+            error("unable to write error log: ", e.getMessage());
+        }
+
+    }
+
+    private void dumpOpenCLKernel(String id, String entryPoint, String log, byte[] source) {
```

Comment:
Is this method used? If yes, what is the difference between this and the other `dumpKernelSource`?

Reply:
sorry, that was a leftover from a refactoring I did. I will remove it